Feat/15/current sensor driver

Mohamed-Diaa-ES opened the pull request:
Task: Current Sesnor
Developer: Mohammed Diaa 
reviewer: @Ahmed-Ashraf750 @Hisham4Ahmed

Commit:
Initializing the ACS712

Commit:
Initializing the ACS712

Commit:
[Current sensor real coding]
[Stating out the files]
[Feat]   #15

Commit:
[Current sensor real coding]
[Stating out the files]
[Feat]   #15

Commit:
[ACS712 Current Sensor Renaming the Files]
[Renaming the Files]
[Feat]   #15

Commit:
[ACS712 Current Sensor Renaming the Files]
[Renaming the Files]
[Feat]   #15

Commit:
[Adding ADC guard]
[Adding ADC guard to ADC program.c to prevent reinitializing]
[Feat]   #15

Commit:
[Current Sensor Finishing Task needs review and
testing]
[
    -Done the APIs
    -Done the header Files
    ]
[Feat]   #15

Commit:
[Current Sensor Finishing Task needs review and
testing]
[
    -Done the APIs
    -Done the header Files
    ]
[Feat]   #15

Commit:
Doing Review Points

Commit:
Doing Review Points

Ahmed-Ashraf750 committed:
[Voltage Driver Files Added]
        [Files for Voltage Driver Implementation added and the Voltage_Interface.h file was finished]
        [Feat]   #14

Ahmed-Ashraf750 committed:
[ Before-Test Voltage Driver ]
        [Voltage Driver Before Modifications related to
        static variable polling Read]
        [Feat]   #14

Ahmed-Ashraf750 committed:
[ADC Initialization Guard]
        [ADC Initialization Guard Was Addded to Prevent Multiple Initializations]   #1
        [a small conflict with current sensor driver may happens] #2
        [Issue Type]   # NumberOfTask

Ahmed-Ashraf750 committed:
[ADC Guard Proper Fix]
        [Forgot to add guard for ADC initialization in the previous commit]
        [Feat]   #14

Ahmed-Ashraf750 committed:
[ADC Guard Conflict Resolved]
        [Resolved the Conflict in ADC Guard By Copying the ADC_Program.c File From Mohamed Diaa branch
        this conflict happened due to that in the current driver the adc guard was added but not merged to the master branch]
        [Feat]   #14

Commit:
[HCurrent_Sensor is done]
[-applied the callback concept to update global
    to stop the polling actions
 -Udpated each functions to be all of O(1)
 ]
[Feat]   #15

Commit:
[HCurrent_Sensor is done]
[-applied the callback concept to update global
    to stop the polling actions
 -Udpated each functions to be all of O(1)
 ]
[Feat]   #15

Commit:
Fixing Doxygen Coments

Commit:
Fixing Doxygen Coments

Ahmed-Ashraf750 committed:
[Voltage Sensor Driver Modifying max Voltage limit]
        [modified max voltage limit from 55V to 300V by changing the resistor values in the voltage divider circuit]
        [feat]   #14

Ahmed-Ashraf750 committed:
[handling the rms in callback]
[implemented the RMS Read in callback function ]
[Feat] #14

Ahmed-Ashraf750 committed:
[Before Handling Heavy ADC ISR Load Voltage Driver]
[this commit holds the working blocking reading instant value and non-working RMS value due to heavy ISR Load in Async Mode]
[Feat]#14

Ahmed-Ashraf750 commented:
@Mohamed-Diaa-ES please cancel this pull request untill the & Heavy ISR issue is fixed

Ahmed-Ashraf750 committed:
debugging

Ahmed-Ashraf750 committed:
Resolving ADC Bug

Ahmed-Ashraf750 committed:
[Pre-Review Voltage Driver Task]
        [Voltage Driver Implementation
        Tested and Verified And Working Perfectly
        only is waiting for Reviewer Approval]
        [Feat]   #14

Ahmed-Ashraf750 committed:
[Handling ADC ISR With no rubber  ]
        [Handled Triggering ADC via Timer1]
        [Feat]   #14

Ahmed-Ashraf750 committed:
[Moved Samples Count to Config File]
        [it may Be Changed By the User Later]
        [Feat]   #14

Commit:
ADC Fixing problem of ISR

Commit:
[Fixing the Bugs in the Sensor]
[
-Changing Timer1 Clock timer to suit the ADC
-Decalring the Function
]
[Feat]   #15

Commit:
Removing the Declaration from the interface and put it in the program.c to precent calling

Commit:
Removing the Declaration from the interface and put it in the program.c to precent calling

Ahmed-Ashraf750 committed:
[Cleaned the Driver Code]
        [Removed the include of LCD Driver from Mcal Layer'
        Used Macro for Timer1 Compare Match Value in Config file instead of Magic Number]
        [Feat]   #14

Commit:
Resolving Some Basic Conflicts

Ahmed-Ashraf750 committed:
Applied Review Notes , [Feat] , #14

Ahmed-Ashraf750 committed:
Handling the instant Value Bug when switching by isr

Commit:
TESTING

Commit:
TESTING

Commit:
TESTING

Commit:
TESTING

Commit:
TESTEING

Commit:
Mergin with my branch

Commit:
Mergin with my branch

Commit:
Mergin with my branch

Commit:
TSTing

Commit:
Editing The data with anohter Driver

Commit:
Correcting Name of config file

Commit:
Editing The value of Sensivity

Commit:
Finishing Testing of the values

Commit:
Ensuring the values

Commit:
Fixing the sensivity Factor

Commit:
Pushing last settings
